src/servers/middlewares/kratos-router.ts

```diff
@@ -37,34 +41,47 @@ kratosRouter.post(
       }
 
       const body = req.body
-      const { identity_id: userId, phone: phoneRaw, schema_id } = body
+      const { identity_id: userId, phone: phoneRaw, schema_id, email } = body
 
-      assert(schema_id === "phone_no_password_v0", "unsupported schema")
+      assert(schema_id === "phone_or_email_password_v0", "unsupported schema")
 
-      if (!phoneRaw || !userId) {
+      if ((!phoneRaw && !email) || !userId) {
         console.log("missing inputs")
         res.status(400).send("missing inputs")
         return
       }
 
-      const phone = checkedToPhoneNumber(phoneRaw)
-      if (phone instanceof Error) {
-        console.log("invalid phone")
-        res.status(400).send("invalid phone")
-        return
-      }
-
       const userIdChecked = checkedToUserId(userId)
       if (userIdChecked instanceof Error) {
         console.log("invalid userId")
         res.status(400).send("invalid userId")
         return
       }
 
-      const account = await createAccountWithPhoneIdentifier({
-        newAccountInfo: { phone, kratosUserId: userIdChecked },
-        config: getDefaultAccountsConfig(),
-      })
+      let account
+      // phone+code flow
+      if (phoneRaw) {
+        const phone = checkedToPhoneNumber(phoneRaw)
+        if (phone instanceof Error) {
+          console.log("invalid phone")
+          res.status(400).send("invalid phone")
+          return
+        }
+        account = await createAccountWithPhoneIdentifier({
+          newAccountInfo: { phone, kratosUserId: userIdChecked },
+          config: getDefaultAccountsConfig(),
+        })
+      } else if (email) {
+        // email+password flow
+        // kratos user likely exists from self registration flow
```

Comment:
```suggestion
        // kratos user exists from self registration flow
```

registration is a private API that should also be called by kratos which mean kratos user is always in the process of been created when this is been called through kratos webhook